pkg/frontend/readpath/queryfrontend/query_frontend_test.go

```diff
@@ -477,3 +478,225 @@ func Test_QueryFrontend_Series_WithLabelNameFiltering(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateStubsForUnsymbolizedProfiles(t *testing.T) {
+	tests := []struct {
+		name          string
+		profile       *profilev1.Profile
+		queries       []*queryv1.Query
+		expectError   bool
+		validateStubs func(t *testing.T, profile *profilev1.Profile)
+	}{
+		{
+			name: "creates stubs for unsymbolized locations",
+			profile: &profilev1.Profile{
+				StringTable: []string{"", "/usr/lib/libjvm.so"},
+				SampleType:  []*profilev1.ValueType{{Type: 1, Unit: 1}},
+				Mapping: []*profilev1.Mapping{
+					{Id: 1, Filename: 1, HasFunctions: false},
+				},
+				Location: []*profilev1.Location{
+					{Id: 1, MappingId: 1, Address: 0xcafebabe, Line: nil},
+					{Id: 2, MappingId: 1, Address: 0xdeadbeef, Line: nil},
+				},
+				Function: []*profilev1.Function{},
+				Sample: []*profilev1.Sample{
+					{LocationId: []uint64{1, 2}, Value: []int64{100}},
+				},
+			},
+			queries: []*queryv1.Query{{QueryType: queryv1.QueryType_QUERY_PPROF}},
```

Comment:
Nit: For unsymbolized locations without an address (0x0), I want to avoid that we output `<file> 0x0`, we should do `<file>` instead.

Reply:
Done in e8f51b370de56140655021014abe790535680482